src/NHibernate/Transaction/AdoNetWithDistributedTransactionFactory.cs

```diff
@@ -90,7 +90,7 @@ public void ExecuteWorkInIsolation(ISessionImplementor session, IIsolatedWork wo
 			}
 		}
 
-		public class DistributedTransactionContext : ITransactionContext, IEnlistmentNotification
+		public class DistributedTransactionContext : ITransactionContext, IEnlistmentNotification, IPromotableSinglePhaseNotification
```

Comment:
Has it really changed anything to add that interface while still enlisting the resource the same way at line 71? I do not think so. I think this change is unnecessary. Please tell me if undoing it causes failure for non-distributed cases.